zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/AuthenticationProvider.java

```diff
@@ -49,6 +52,19 @@ public interface AuthenticationProvider {
      */
     KeeperException.Code handleAuthentication(ServerCnxn cnxn, byte[] authData);
 
+    /**
+     * This method is called when admin server command passes authentication data for this
+     * scheme.
+     *
+     * @param request
+     *                the request that contains the authentication information.
+     * @param authData
+     *                the authentication data received.
+     * @return Ids
+     *                the list of Id. Empty list means not authenticated
+     */
+    List<Id> handleAuthentication(HttpServletRequest request, byte[] authData);
```

Comment:
`default` to empty list and thus reduce some impls?

Reply:
Yep, good idea. Changed.